src/pages/Account/CodeOfConduct.tsx

```diff
@@ -27,21 +26,21 @@ export interface CodeOfConductQuestion {
 
 const CodeOfConduct: React.FunctionComponent<PropsType> = ({ location }) => {
   useConnectCurrentVenue();
-
-  const history = useHistory();
   const { user, venue } = useSelector((state: any) => ({
     user: state.user,
     venue: state.firestore.data.currentVenue,
   }));
-  const { venueId } = getQueryParameters(location.search);
+  const { venueId, eventId } = getQueryParameters(location.search);
   const { register, handleSubmit, errors, formState, watch } = useForm<
     CodeOfConductFormData
   >({
     mode: "onChange",
   });
   const onSubmit = async (data: CodeOfConductFormData) => {
     await updateUserProfile(user.uid, data);
-    history.push(`/${venueId ? `venue/${venueId}${location.search}` : ""}`);
+    window.location.assign(
+      `/${venueId && eventId ? `venue/${venueId}/event/${eventId}` : ""}`
+    );
```

Comment:
why bypass router? there should be a function to replace assign within the react-router-dom